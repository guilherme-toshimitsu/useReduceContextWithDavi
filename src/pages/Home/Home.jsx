import React from 'react';
import PropTypes from 'prop-types';
import { Button } from '@material-ui/core';

import Api from '@api';

const Home = ({ history }) => {
  const handleClick = () => {
    Api.seller.getAll().then(data => true);
  };
  return (
    <div>
      <Button onClick={() => history.push('/seller')} color="primary" variant="contained">
        GoToSeller
      </Button>
      <Button onClick={() => history.push('/seller/123456789')} color="primary" variant="contained">
        goToSellerId
      </Button>
      <div style={{ marginTop: '150px' }}></div>
      <Button onClick={handleClick}>APICALL</Button>
    </div>
  );
};

Home.propTypes = {
  history: PropTypes.object,
};

export default Home;
