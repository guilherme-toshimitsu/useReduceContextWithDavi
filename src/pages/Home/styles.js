const styles = {
  root: {
    width: '100%',
  },
  tableWrapper: {
    maxHeight: 440,
    overflow: 'auto',
    display: 'flex',
  },
  titleTableSeller: {
    backgroundColor: '#f4f4f4',
    textAlign: 'center',
  },
  firstTitleTableSeller: {
    backgroundColor: '#f4f4f4',
    paddingLeft: 32,
    color: 'rgba(0, 0, 0, 0.87)',
    fontWeight: 500,
    lineHeight: '1.5rem',
  },
  firstColumnTableSeller: {
    paddingLeft: 32,
  },
  bodyTableSeller: {
    textAlign: 'center',
    // width: 78,
    height: 18,
    fontSize: 13,
    textAlign: 'center',
    color: '#404040',
  },
  paginator: {
    display: 'flex',
    justifyContent: 'center',
    color: '#0086ff',
    disabled: {
      color: '#0086ff',
      borderColor: '#ddd',
    },
  },
  pagination: {
    color: '#0086ff',
    border: 'solid 1px #e5e5e5',
    display: 'inline-block',
    textAlign: 'center',
    padding: 5,
    margin: '20 0',
    fontSize: 14,
    borderRadius: 4,
    disabled: {
      linkDisabled: {
        color: '#777',
        borderColor: '#ddd',
        cursor: 'not-allowed',
      },
    },
    active: {
      linkActive: {
        zIndex: 3,
        color: '#00a9d7',
        backgroundColor: '#f5f5f5',
        borderColor: '#d6d6d6',
        cursor: 'default',
      },
    },
    li: {
      display: 'inline',
      '&:first-child': {
        linkFirstChild: {
          marginLeft: 0,
          borderBottomLeftRadius: 4,
          borderTopLeftRadius: 4,
        },
      },
      '&:last-child': {
        linkLastChild: {
          borderBottomRightRadius: 4,
          borderTopRightRadius: 4,
        },
      },
      linkLi: {
        position: 'relative',
        float: 'left',
        padding: '6 12',
        lineheight: '1.4385',
        textDecoration: 'none',
        border: '1px solid #ddd',
        backgroundColor: '#fff',
        color: '#00b1e1',
        marginLeft: -1,
      },
    },
  },
  actions: {
    active: { color: '#0086ff' },
  },
};

export default styles;
