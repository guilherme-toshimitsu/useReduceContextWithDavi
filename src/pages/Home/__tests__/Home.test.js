import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Home from '../';

describe('<Home> Home Page Container', () => {
  it('renders', () => {
    const container = render(<Home />);
    expect(container).toMatchSnapshot();
  });
});
