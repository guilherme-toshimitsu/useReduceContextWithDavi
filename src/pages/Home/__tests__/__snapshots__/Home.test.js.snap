// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home> Home Page Container renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            GoToSeller
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            goToSellerId
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          style="margin-top: 150px;"
        />
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            APICALL
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          GoToSeller
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          goToSellerId
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        style="margin-top: 150px;"
      />
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          APICALL
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
